core/domain/question_domain.py

```diff
@@ -209,6 +211,26 @@ def _convert_state_v28_dict_to_v29_dict(cls, question_state_dict):
         question_state_dict['solicit_answer_details'] = False
         return question_state_dict
 
+    @classmethod
+    def _convert_state_v29_dict_to_v30_dict(cls, question_state_dict):
+        """Converts from version 29 to 30. Version 30 add image assets
+        in the question_state_dict.
+
+        Args:
+            question_state_dict: dict. The dict representation of
+                question_state_data.
+
+        Returns:
+            dict. The converted question_state_dict.
+        """
+        image_counter = 0
+        (html_validation_service.
```

Comment:
What are we doing here, can you explain?

Reply:
adding ```image_counter``` in ```question_state_dict```.